lib/entice/logic/MapInstance.ex

```diff
@@ -0,0 +1,78 @@
+defmodule Entice.Logic.MapInstance do
+  alias Entice.Entity
+  alias Entice.Logic.MapInstance
+
+
+  defstruct(
+    players: [],
+    npcs: [],
+    map: nil)
+
+  def join(entity, player_entity) do
+    Entity.call_behaviour(entity, MapInstance.Behaviour, {:map_instance_entity_join, player_entity})
+  end
+
+  def leave(entity, player_entity) do
+    Entity.call_behaviour(entity, MapInstance.Behaviour, {:map_instance_entity_leave, player_entity})
+  end
+
+  def register(entity, map, players=[], npc_info=[]),
```

Comment:
afaics `players=[]` and the other param is two things: incorrect syntax and not really giving us any clue of what we expect to get passed in
default values can be with the `param \\ :default_value` syntax
however, we need to make sure that functions with default values do not clash in arity with other functions